Android test variant support

Tanish-Ranjan opened the pull request:
Addressing https://github.com/microsoft/build-server-for-gradle/issues/183, this PR discovers all the test variants and adds them to the source sets.

Tanish-Ranjan commented:
This has been branched from android-support branch. I will rebase once https://github.com/microsoft/build-server-for-gradle/pull/173 is merged.

Tanish-Ranjan committed:
Added test variants to the source sets.

Changes:
- Updated AndroidUtils to populate all fields individually so failure in retrieval of one property won't lead to the collapse of whole source set.
- Added android resource directories only if the variant is not a unit test
- Updated GradleApiConnectorTest#testAndroidSourceSets to consider the all the test variants and their dependencies
- Addressed checkstyle issues

Tanish-Ranjan commented:
Tested the changes with default test project by integrating with [vscode-gradle](https://github.com/microsoft/vscode-gradle) extension. The test variants were retrieved successfully.

jdneo commented on plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java:
```diff
@@ -156,8 +163,57 @@ private static GradleSourceSet convertVariantToGradleSourceSet(Project project,
+      gradleSourceSet.setArchiveOutputFiles(new HashMap<>());
+
+      // has tests
+      addTests(gradleSourceSet, project, variant, sourceSets);
```
Why `addTests()` put inside `convertVariantToGradleSourceSet()`? Can we handle them separately?

jdneo commented on plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java:
```diff
@@ -116,9 +117,15 @@ public static List<GradleSourceSet> getBuildVariantsAsGradleSourceSets(Project p
    */
-  @SuppressWarnings("unchecked")
-  private static GradleSourceSet convertVariantToGradleSourceSet(Project project, Object variant) {
+  private static GradleSourceSet convertVariantToGradleSourceSet(
```
We can change the function to return a collection of source set, and add them out of the function

Tanish-Ranjan replied:
Resolved in https://github.com/microsoft/build-server-for-gradle/pull/194/commits/736a14ca181bcf75ac0c11fa65d4b691192cf3c8

Tanish-Ranjan committed:
Refactored AndroidUtils test source set retrieval.

Changes:
- AndroidUtils#convertVariantToGradleSourceSet now returns a list of source sets rather than a single source set.
- AndroidUtils#addTests now returns a list of test source sets.

Tanish-Ranjan committed:
Refactored Android variants retrieval.

jdneo commented:
Thank you for your contribution! @Tanish-Ranjan
